src/Metable.php

```diff
@@ -75,6 +75,20 @@ public function setMeta(string $key, $value): void
         }
     }
 
+    /**
+     * Add or update many `Meta` values.
+     *
+     * @param array|Traversable $array
+     *
+     * @return void
+     */
+    public function setManyMeta($array): void
+    {
+        foreach ($array as $key => $value) {
+            $this->setMeta($key, $value);
+        }
```

Comment:
Same thing, though a little trickier to bulk insert efficiently. Laravel 8.0 supports `$queryBuilder->upsert()`, though the current version of this package still supports laravel 6.0+. We would need two implementations, or higher requirements for this specific method.

Reply:
We could certainly `version_compare()` and use `upsert()` when possible, and use `updateOrCreate(['key' => $key], ['value' => $value])` otherwise. It's effectively doing the same thing as `setMeta()` though, minus updating the cached relationship. If the relationship is already loaded, `setMeta()` actually does less than `updateOrCreate()`, since you don't have the lookup query to determine existence.

The more I look at it, the more I think that with the way you're utilizing the cached relationship on `$indexedMetacollection`, unless you can use `upsert()`, what's the harm in doing it the way that I did it? You're always going to have to...

1. Figure out if the meta key exists
2. Create a new `Meta` _or_ change `$meta->value`
3. `save()` the `Meta` instance
4. Save it to the cached relationship

What else do you have in mind for how that could be optimized?

Reply:
`version_compare()` is for PHP version, not Laravel version. You could use `Illuminate\Foundation\Application::version()` for that instead.

Unfortunately, `updateOrCreate()` can only take multiple records as of Laravel 8.0, so that's no better. We might have to live with < 8.0 using your existing loop and >= 8.0 using upsert. 

To your point about needing to update the indexed cache as well, the easiest approach would be to do the upsert and then dump the cache. The next time a read method is called it will reload it with the up-to-date data.

However, I just realized that we are missing one critical piece for upsert: we need a unique index of `metable_type, metable_id, key`. That should probably be added anyways, though it will require a major version bump to add a new migration. Note: this could replace the `metable_type, metable_id` index, since they start with the same columns.

Reply:
I'm still planning on making these changes, don't give up on me just yet 😂 I run an ecommerce software company and Q4 is full of 18+ hour days. Fun!

I just wanted to comment that a unique index on `metable_type, metable_id, key` IMO is 100% necessary (that and using methods like `updateOrCreate()` when inserting/updating the meta key+value). I was looking through some of my models the other night and have found that as a result of having a heavily queued-job-based architecture, I have _a lot_ of meta key duplication. It's easy to do - for example: dispatch two different queued jobs at the same time that both result in a `setMeta()` on the same model with the same key.

Here's an example of the meta on one of my models:

![image](https://user-images.githubusercontent.com/748444/102302516-6fa75600-3f16-11eb-91ff-8978066f02d5.png)

Reply:
We could do something that's easier than checking versions: `method_exists()`

About as foolproof as they come 😃